test/extended/two_node/utils/apis/baremetalhost.go

```diff
@@ -58,3 +70,67 @@ func GetBMH(oc *exutil.CLI, bmhName, namespace string) (*metal3v1alpha1.BareMeta
 
 	return &bmh, nil
 }
+
+// RotateNodeBMCPassword discovers the BMC Secret for the given node,
+// rotates its "password" key and returns (namespace, secretName, originalPassword).
+func RotateNodeBMCPassword(kubeClient kubernetes.Interface, node *corev1.Node) (string, string, []byte, error) {
+	ctx := context.Background()
+	secretClient := kubeClient.CoreV1().Secrets(BMCSecretNamespace)
+
+	secrets, err := secretClient.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return "", "", nil, fmt.Errorf("failed to list secrets: %w", err)
+	}
+
+	var target *corev1.Secret
+	for _, s := range secrets.Items {
+		if strings.Contains(s.Name, node.Name) &&
+			strings.Contains(s.Name, "bmc") {
+			target = &s
+			break
+		}
+	}
+
+	if target == nil {
+		return "", "", nil, fmt.Errorf("no BMC secret found for node %s", node.Name)
+	}
+
+	// Save original password
+	original := target.Data["password"]
```

Comment:
Very small nitpick, lets have this key be a const since we're using it in multiple spots, that should avoid accidental mis-spellings.

```go
const (
	BMCSecretNamespace = "openshift-machine-api"
    secretsDataPasswordKey = "password"
)
```